extensions/amp-social-share/0.1/amp-social-share-config.js

```diff
@@ -73,4 +74,10 @@ const BUILTINS = {
       url: 'CANONICAL_URL',
     },
   },
+  whatsapp: {
+    shareEndpoint: 'whatsapp://send',
+    defaultParams: {
+      text: 'TITLE - CANONICAL_URL',
```

Comment:
**For multiligual urls** (i.e. urls, where say Hindi or German scripts are present in form of urlecoded characters):
* Though this will work, but will not show the _cool snippet_ which appears within whatsapp; As it appears from server logs that WhatsApp further url-encode appended link, resulting into twice url-encoded link hitting the server inturn resulting to 404-not found errors on wordpress and many content platforms.

I am appending a simple hack to fix this:
* set `data-param-text="Some text - URLDECODED(_canonical_url_)` where _canonical_url_ is variable with valid CANONICAL_URL URLDECODED is a function that url-decodes a string, normally with a library implementation in all languages

PS: 
* case live on the edge of multiple standards by browsers as well as content platforms wrt to handling multilingual urls
* Test works fine with Chrome, Firefox, Edge